### PR TITLE
feat: facades.Orm supports cursor

### PR DESCRIPTION
**Pull request by kkumar-gcc**
Closes https://github.com/goravel/goravel/issues/95

## 📑 Description
This PR adds support for the `Cursor` method in the query, enabling efficient handling of large result sets without loading all data into memory.

## ✅ Checks
- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: database/gorm/query.go [hunk 1]
```diff
@@ -28,6 +29,10 @@ type QueryImpl struct {
 	withoutEvents bool
 }
 
+type CursorImpl struct {
```
Comment (hwbrzzl) [inline database/gorm/query.go:32]: S: Create a new file for `CursorImpl`.

File: database/gorm/query.go [hunk 2]
```diff
@@ -43,6 +48,43 @@ func NewQueryImpl(ctx context.Context, config config.Config, gorm Gorm) (*QueryI
...
+		rows, err := r.instance.Rows()
+		if err != nil {
+			return
+		}
+
+		for rows.Next() {
+			val := make(map[string]any)
+			err := r.instance.ScanRows(rows, val)
+			if err != nil {
+				err = rows.Close()
```
Comment (hwbrzzl) [inline database/gorm/query.go:64]:
```suggestion
```

File: database/gorm/query.go [hunk 3]
```diff
@@ -43,6 +48,43 @@ func NewQueryImpl(ctx context.Context, config config.Config, gorm Gorm) (*QueryI
...
 
+func (r *QueryImpl) Cursor() (chan ormcontract.Cursor, error) {
+	var err error
+	cursorChan := make(chan ormcontract.Cursor)
+	go func() {
+		rows, err := r.instance.Rows()
+		if err != nil {
+			return
+		}
+
```
Comment (hwbrzzl) [inline database/gorm/query.go:59]:
```suggestion
defer rows.Close()
```

**Review by hwbrzzl — COMMENTED** [review]
Good job, bro, that's what we want. Please add some test cases after I solve the timestamp problem in `mapstructure`, then this's PR can be merged.

File: database/gorm/query_test.go [hunk 4]
```diff
@@ -2458,6 +2458,29 @@ func (s *QueryTestSuite) TestSum() {
...
+			s.Nil(query.Create(&user))
+			s.True(user.ID > 0)
+
+			user1 := User{Name: "cursor_user", Avatar: "cursor_avatar1"}
+			s.Nil(query.Create(&user1))
+			s.True(user1.ID > 0)
+
+			users, err := query.Model(&User{}).Where("name = ?", "cursor_user").Cursor()
+			s.Nil(err)
+			for row := range users {
```
Comment (hwbrzzl) [inline database/gorm/query_test.go:2474]: S: Please add a length assert for `row`. Such as: 
```
var len int
for row := range users {
  len++
}
s.Equal(2, len)
```

**Review by hwbrzzl — COMMENTED** [review]
Good job 👍

File: database/gorm/query.go [hunk 5]
```diff
@@ -81,6 +82,34 @@ func (r *QueryImpl) Create(value any) error {
...
+		if err != nil {
+			return
+		}
+		defer func(rows *sql.Rows) {
+			err := rows.Close()
+			if err != nil {
+
+			}
+		}(rows)
+
```
[the remark this answers is not in the document]
Comment (kkumar-gcc) [reply database/gorm/query.go:99]: Well this was giving me below warning that's why i changed it.
image

**Review by hwbrzzl — APPROVED** [review]
LGTM

File: database/gorm/query.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (hwbrzzl) [reply database/gorm/query.go]: That's ok here.

**Pull request merged**